### ISSUE TITLE
Training Time

### ISSUE THREAD

**Issue by AmingWu** [shvdiwnkozbw/Multi-Source-Sound-Localization#1]
How long does your model need to train? How much GPU memory does your model need?

**Comment by shvdiwnkozbw** [comment]
We strongly recommend to use the simplfied version in folder ./new/, which is much easier to train and more computationally efficient.
For this version, the small batchsize 16 works fine, and only requires about 2G GPU memory with input frame 256x256, and audio length 5 seconds. But larger batchsize eg 64 is preferred. Normally, one single 1080Ti is enough. And the whole training process takes about 2h for 10K training data, proportional to training data size.

**Comment by AmingWu** [comment]
OK, Thank you very much. I will try to reproduce your experiments.

**Comment by AmingWu** [comment]
![QQ截图20200717130027](https://user-images.githubusercontent.com/38445175/87750281-98bf9f80-c82d-11ea-925e-18a76fbe4fe3.png)
This is my download data. How to get it, i.e., /media/ruiq/Data/AudioVisual/data/Flickr/Spec.h5?

**Comment by shvdiwnkozbw** [comment]
You can use python library librosa to load and process the audio, either wav or mp3 file. Concretely, use librosa.load() to load the audio file at a specific sample rate, eg 22.05 kHz, and librosa.feature.melspectrogram() to transform into mel-spectrogram. In our implementation, we use 64 frequency bins from a window of 40ms every 20ms.
An example is shown as:

wav, sr = librosa.load(wavefilepath, sr=SAMPLE_RATE)
logmel_spec = np.log(librosa.feature.melspectrogram(wav, sr, n_fft=int(0.04*sr), hop_length=int(0.02*sr), n_mels=64) + np.spacing(1)).T

Each generated log-mel-spectrogram is of size TxF, and for all in the dataset, we concate into the Nx1xTxF to be stored in the final spec.h5

**Comment by GenjiB** [comment]
Hi, 

Could you provide Spec.h5//Video.h5/labels_a.npy/labels_v.npy or the preprocess codes for easy reproducing results?

Thanks

**Comment by shvdiwnkozbw** [comment]
Thanks for your suggestion, we have updated the preprocessing code.
The detailed preprocessing code and classification pseudo label generation can be referred to ```./prepare_data``` and ```./mobile_crnn/generate_data.py```. The pseudo labels for images are similar to the audio, but replace the model with ImageNet pretrained backbones, and use ```./utils/cluster_v3.npy``` for aggregation. And it is also practical to use the same pseudo labels for both images and audios, since the ImageNet labels cannot be well aligned with 7 general categories in AudioSet, introducing extra noise. There is only slight difference in the choice of label assignment. For datasets like AudioSet, directly using videos tags as classification labels.

**Comment by poult-lab** [comment]
Hello author, thanks so much to your contribution of paper. But I still can't find where is '/media/ruiq/Data/AudioVisual/data/Flickr/Spec.h5'.
when I run this code I got OSError : Unable to open file (unable to open file: name = '/media/ruiq/Data/AudioVisual/data/Flickr/Spec.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)

**Comment by champagne-yellow** [comment]
Hi, shvdiwnkozbw. Nice work, but I am very confused. In **prepare_data/data.py** :
```python
    imgs.append(img)
    audios.append(logmel_spec)
    
with h5py.File('path/to/img_h5', 'w') as f:
    f['video'] = np.array(imgs)
with h5py.File('path/to/audio_h5', 'w') as f:
    f['audio'] = np.array(logmel_spec)
```
Should it be changed to：
```python
    imgs.append(img)
    audios.append(logmel_spec)
    
with h5py.File('path/to/img_h5', 'w') as f:
    f['video'] = np.array(imgs)
with h5py.File('path/to/audio_h5', 'w') as f:
    f['audio'] = np.array(audios)
```
Did I get it wrong?
